pkg/controller/worker/machines.go

```diff
@@ -376,3 +379,20 @@ func computeIAMInstanceProfile(workerConfig *awsapi.WorkerConfig, infrastructure
 
 	return nil, fmt.Errorf("unable to compute IAM instance profile configuration")
 }
+
+func computeInstanceMetadata(workerConfig *awsapi.WorkerConfig) map[string]interface{} {
+	res := make(map[string]interface{})
```

Comment:
we should default `httpPutResponseHopLimit` to 2 here.

Reply:
If we default it to 2, then we change the default behavior to enable IMDSv2  by default and this is not what we should do for now. We can in the future consider this with proper "breaking" announcements, but let's not change the defaults in this iteration.

Reply:
okay , would adding a `TODO` here make sense ? because in the docString of `HTTPResponseHopLimit` its written that the hop limit is 2, so someone reading the code might get confused.

Reply:
This was a problem with the documentation I that I forgot to remove. I removed the mention of the defaulting - thank you for this catch.

The defaulting only makes sense with the old set of flags that I had exposed.